test/test_improved_search_truncation.js

```
@@ -1,60 +1,115 @@
-        } else if (result.content[0].text.length > 900000) {
-            console.log('⚠️  Results close to limit but acceptable');
+        console.log('Initial result type:', typeof initialResult.content[0].text);
+        console.log('Initial result length:', initialResult.content[0].text.length);
+        console.log('Final result type:', typeof finalResult.content[0].text);
+        console.log('Final result length:', finalResult.content[0].text.length);
+        
+        const totalLength = initialResult.content[0].text.length + finalResult.content[0].text.length;
+        const apiLimit = 1048576; // 1 MiB - use consistent constant
+        
+        // Check if we're within the safe limits using single source of truth
+        if (totalLength > apiLimit) {
```

Biome (2.1.2)

[error] This variable is used before its declaration.

The variable is declared here:


(lint/correctness/noInvalidUseBeforeDeclaration)